cypress/visual/AppNavigationItem.cy.ts

```diff
@@ -0,0 +1,36 @@
+import { mount } from 'cypress/vue2'
+import Vue, { defineComponent } from 'vue'
+import VueRouter from 'vue-router'
+
+import NcAppNavigationItem from '../../src/components/NcAppNavigationItem/NcAppNavigationItem.vue'
+
+describe('NcAppNavigationItem', () => {
+	it('With router link is primary on active', () => {
+		const RouterComponent = defineComponent({
+			template: '<div style="width: 300px; background: white;"><NcAppNavigationItem name="Home" to="/" :editable="true" /><NcAppNavigationItem name="Foo" to="/foo" :editable="true" /></div>',
+			components: { NcAppNavigationItem },
+		})
+
+		const router = new VueRouter({
+			routes: [
+				{ path: '/' },
+				{ path: '/foo' },
+			],
+		})
+
+		Vue.use(VueRouter)
+
+		mount(RouterComponent as never, {
+			extensions: {
+				plugins: [router],
+			},
+			router,
+		})
+
+		cy.contains('.app-navigation-entry', 'Home').compareSnapshot('NcAppNavigationItem-primary-on-active-route')
+		cy.contains('.app-navigation-entry', 'Foo').compareSnapshot('NcAppNavigationItem-tertiary-non-active-route')
+
+		cy.contains('.app-navigation-entry', 'Home').find('.app-navigation-entry__actions').click()
+		cy.get('.app-navigation-entry--editing').compareSnapshot('NcAppNavigationItem-primary-on-active-route-menu-active')
```

Comment:
I'm always afraid of screenshot testing 🙈 
Too many factors changes pixels...

Reply:
So, CI says 

> Error: The "NcAppNavigationItem-primary-on-active-route" image is different. Threshold limit exceeded! differently

And we even cannot look at it on the GitHub to see if the test works differently here or the component doesn't work as expected.

Reply:
currently trying to fix this, this happens if I create the snapshots on my local machine. CI uses ubuntu which has different color interpolation / aliasing. (not sure how this is called).

Reply:
> I'm always afraid of screenshot testing 🙈 Too many factors changes pixels...

Yes but hard to otherwise test visual stuff :wink:

Reply:
We have the same issue for the `NcAppSidebar` tests. But you can download the snapshots generated by the cypress action and check them in as the base snapshots. That makes the tests pass on GitHub, but they of course still fail locally than. But that might be acceptable.

That being said, there seems to be something else wrong, that just the aliasing. These are the remote snapshots:
![NcAppNavigationItem-primary-on-active-route](https://github.com/nextcloud-libraries/nextcloud-vue/assets/2496460/8e965c4b-f6d4-4499-854b-a79df76afe4f)
![NcAppNavigationItem-primary-on-active-route-menu-active](https://github.com/nextcloud-libraries/nextcloud-vue/assets/2496460/4f5b5643-5b51-4871-9ce8-ff99c77ac4c6)
![NcAppNavigationItem-tertiary-non-active-route](https://github.com/nextcloud-libraries/nextcloud-vue/assets/2496460/976cf90c-5039-4089-8c24-0b68d8bca996)

Reply:
@raimund-schluessler but all tests have passed, see changes for the checked-in snapshots which look fine.

BTW to create working snapshots I use something like:
`docker run -it -v .:/app --entrypoint /bin/sh -w app cypress/included:13.3.1 npm run cypress:update-snapshots`

Reply:
> @raimund-schluessler but all tests have passed, see changes for the checked-in snapshots which look fine.
> 
> BTW to create working snapshots I use something like: `docker run -it -v .:/app --entrypoint /bin/sh -w app cypress/included:13.3.1 npm run cypress:update-snapshots`

Right, sorry. I looked at the `diff`s instead of the actual ones :see_no_evil: All good in that case.

Reply:
Ok :)
Feel free to leave a review

Reply:
> BTW to create working snapshots I use something like: `docker run -it -v .:/app --entrypoint /bin/sh -w app cypress/included:13.3.1 npm run cypress:update-snapshots`

Could be nice to put this in the readme.

Reply:
> Yes but hard to otherwise test visual stuff 😉

Manually, having more examples in the docs =D

Having tests is not the final goal, they should not make more troubles than profits.

From my experience, snapshot and visual regression tests often make more trouble, because they require reproducible environment (e.g. launch Cypress GUI to debug results in a different screenshot than launch without GUI), has no result in the CI log, hard to tell that goes wrong when there is no obvious difference, and developers always have an option to say "just a valid change, let's update snapshot".